kubernetes/kubernetes.go

```diff
@@ -30,25 +32,34 @@ import (
 
 type CronSchedule string
 
-const MaxNameLength = 53
+const MaxJobNameLength = 50
 
-func GetJobName(id metadata.ResourceID, image string) string {
-	resourceName := fmt.Sprintf("%s-%s-%s", id.Type, id.Name, id.Variant)
-	if len(resourceName) > MaxNameLength {
-		resourceName = resourceName[:MaxNameLength]
+// CreateJobName Only the first value in prefixes will be used.
+func CreateJobName(id metadata.ResourceID, prefixes ...string) string {
+
+	// use timestamp to generate unique job name
+	timestamp := time.Now().UnixNano() / int64(time.Millisecond) // get current time in milliseconds
+	timestampBase36 := strconv.FormatInt(timestamp, 36)          // convert timestamp to base36
+
+	jobNameBase := fmt.Sprintf("%s-%s-%s-%s", id.Type, id.Name, id.Variant, timestampBase36)
+
+	// if jobPrefix is provided, prepend it to jobNameBase
+	if len(prefixes) > 0 && prefixes[0] != "" {
+		jobNameBase = fmt.Sprintf("%s-%s", prefixes[0], jobNameBase)
 	}
-	jobName := strings.ReplaceAll(resourceName, "_", ".")
-	removedSlashes := strings.ReplaceAll(jobName, "/", "")
-	removedColons := strings.ReplaceAll(removedSlashes, ":", "")
-	MaxJobSize := 63
-	lowerCase := strings.ToLower(removedColons)
-	jobNameSize := int(math.Min(float64(len(lowerCase)), float64(MaxJobSize)))
-	lowerName := lowerCase[0:jobNameSize]
-	return lowerName
-}
 
-func GetCronJobName(id metadata.ResourceID) string {
-	return strings.ReplaceAll(fmt.Sprintf("featureform-%s-%s-%s-%d", strings.ToLower(string(id.Type)), strings.ToLower(id.Name), strings.ToLower(id.Variant), id.Type), "_", ".")
+	// clean up job name for k8s
+	replacer := strings.NewReplacer("_", ".", "/", "", ":", "")
+	jobNameBase = replacer.Replace(jobNameBase)
+
+	lowerCased := strings.ToLower(jobNameBase)
+
+	// truncate job name to MaxJobNameLength characters
+	if len(lowerCased) > MaxJobNameLength {
```

Comment:
Feel like the timestamp should be after truncation, otherwise we might cut it off entirely